oxd-server/src/main/java/org/xdi/oxd/server/op/GetAccessTokenByRefreshTokenOperation.java

```diff
@@ -66,6 +75,22 @@ public IOpResponse execute(GetAccessTokenByRefreshTokenParams params) throws Exc
         throw HttpException.internalError();
     }
 
+    private void handleInvalidRefreshToken(TokenResponse tokenResponse) {
+
+        final TokenErrorResponseType errorType = tokenResponse.getErrorType();
+        final ErrorResponseCode errorCode;
+
+        if (errorType == INVALID_GRANT) {
+            errorCode = INVALID_REFRESH_TOKEN;
+
+        } else {
+            errorCode = INTERNAL_ERROR_NO_PARAMS;
+        }
+
+        throw new HttpException(errorCode);
```

Comment:
right before exiting from the method please log actual errorType and result `errorCode`, so when someone send to us log file we see exact mapping. It's obvious here but during time when it will be changed, it's going to be not as obvious.